recipes/mcstas/install-files-core.sh

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bash
+
+#Make sure we fail hard in case of trouble:
+set -e
+set -u
+set -x
+
+SRCDIR="$PWD/src"
+
+#A few (silent) sanity checks that variables are set and meaningful:
+test -d "${PREFIX}"
+test -d ${SRCDIR}
+test -f ${SRCDIR}/CMakeLists.txt
+test -n "${PKG_VERSION}"
+
+for i in $(seq 1 100000); do
+    #Find a unique build dir:
+    BLDDIR="$PWD/build_mcstas_core_${i}"
+    if [ ! -d "${BLDDIR}" ]; then
+        break
+    fi
+done
+
+mkdir "${BLDDIR}"
+cd "${BLDDIR}"
```

Comment:
I'm curious what the point of creating a `BLDDIR` is? It doesn't seem to be specified elsewhere and the build happens in an isolated environment already so, it doesn't appear necessary?

NB: I don't have merge rights - just a casual observer.

Reply:
Thanks for the feedback/question @dhirschfeld! Well, in my experience it always pays off to have the CMake build dir completely clean and isolated from the source directory, otherwise there might be hard to debug issues coming up some day. And in fact, the source code is located in a sub-directory of the working directory (`./src/`), so it is indeed not completely clean (imagine if CMake some day started using a cache file named `src`). And as lines 16-22 testify, we did actually run into clashy not-completely-clean issues. With the lines you comment on, we are iron-clad sure to have a nice clean directory, even if something would change in either CMake or conda-forge code that we would not be aware of. Seems like a small price to pay :-)

Reply:
Why not just `mkdir build_mcstas_core`?

Since the build is in an isolated environment this folder will only exist if it exists in the upstream repository. It seems unnecessary to check folder names first?

Reply:
Well, that was what I did the first time around! :-)

Then the tests failed with "directory already exists", and I started to wonder if perhaps some of the smart conda-magic was trying to use the same build dir to somehow reuse the CMake cache. Since the main mcstas developer claim to not support such usage, I came up with the logic above.